File: atcoder/abc/abc361/e/main.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long

struct WeightedGraph{
  
};

template< typename T >
pair< T, int > dfs(const WeightedGraph< T > &g, int idx, int par) {
  pair< T, int > ret(0, idx);
  for(auto &e : g[idx]) {
    if(e.to == par) continue;
    auto cost = dfs(g, e.to, idx);
    cost.first += e.cost;
    ret = max(ret, cost);
  }
  return ret;
}

template< typename T >
T tree_diameter(const WeightedGraph< T > &g) {
  auto p = dfs(g, 0, -1);
  auto q = dfs(g, p.second, -1);
  return (q.first);
}

int main(){
  int N;
  cin >> N;
  vector<pair<ll, int>> G(N, )
}

File: atcoder/abc/abc361/e/main.py
N = int(input())
G = [[] for _ in range(N)]
ans = 0
import sys
sys.setrecursionlimit(10**7)
for _ in range(N - 1):
  A, B, C = map(int, input().split())
  A -= 1
  B -= 1
  G[A].append((B, C))
  G[B].append((A, C))
  ans += C*2

import pypyjit
pypyjit.set_param('max_unroll_recursion=-1')
def dfs(x, par):
  ret = 0
  idx = x
  for nx, c in G[x]:
    if nx == par: continue
    cost, retx = dfs(nx, x)
    cost += c
    if cost > ret:
      idx = retx
      ret = cost
  return ret, idx

p, x = dfs(0, -1)
c, _ = dfs(x, -1)
print(ans - c)